daffodil-runtime1/src/main/scala/org/apache/daffodil/infoset/SAXInfosetOutputter.scala

```diff
@@ -0,0 +1,266 @@
+   * @return true on sucess, false if there was an error and Daffodil should stop all
+   *         future calls to the InfosetOutputter
+   */
+  override def startComplex(diComplex: DIComplex): Boolean = {
+    val contentHandler = xmlReader.getContentHandler
+    val ns = if (diComplex.erd.thisElementsNamespace.isNoNamespace) "" else diComplex.erd.thisElementsNamespace.toString
+    val prefix = if(diComplex.erd.thisElementsNamespacePrefix == null) "" else  diComplex.erd.thisElementsNamespacePrefix
+    val elemName = diComplex.erd.name
+    val qName = if (prefix == "") elemName else s"$prefix:$elemName"
+
+    try {
+      val nsbStart = diComplex.erd.minimizedScope
+      val nsbEnd = if (diComplex.isRoot) scala.xml.TopScope else diComplex.diParent.erd.minimizedScope
+      var n = nsbStart
+      while( n != nsbEnd) {
+        val prefix = if(n.prefix == null) "" else n.prefix
+        val uri = if(n.uri == null) "" else n.uri
+        contentHandler.startPrefixMapping(prefix, uri)
+        n = n.parent
+      }
+
+      val attrs = if(isNilled(diComplex)) {
+        createNilAttribute()
+      } else {
+        new AttributesImpl()
+      }
+
+      contentHandler.startElement(ns, elemName, qName, attrs)
+      true
+    } catch {
+      case _: SAXException => false
+    }
+  }
+
+  /**
+   * Called by Daffodil internals to signify the end of a complex element.
+   *
+   * @param diComplex the complex element that is ended. Various fields of
+   *                  DIComplex can be accessed to determine things like the
+   *                  nil, name, namespace, etc.
+   * @return true on sucess, false if there was an error and Daffodil should stop all
+   *         future calls to the InfosetOutputter
+   */
+  override def endComplex(diComplex: DIComplex): Boolean = {
+    val contentHandler = xmlReader.getContentHandler
+    val ns = diComplex.erd.thisElementsNamespace
+    val nsb = diComplex.namedQName
+    try {
+      contentHandler.endElement(ns, nsb.local, nsb.toQNameString)
+
+      val nsbStart = diComplex.erd.minimizedScope
+      val nsbEnd = if (diComplex.isRoot) scala.xml.TopScope else diComplex.diParent.erd.minimizedScope
+      var n = nsbStart
+
+      // we store the the prefixes in the same order as we started them so we can
+      // reverse them and end them in the enclosing order
+      var prefixes: Seq[String] = Seq()
+      while( n != nsbEnd) {
+        val prefix = if(n.prefix == null) "" else n.prefix
+        prefixes :+= prefix
+        n = n.parent
+      }
+      prefixes.reverse.foreach(contentHandler.endPrefixMapping)
+      true
+    } catch {
+      case _: SAXException => false
+    }
+  }
+
+  private def createNilAttribute(): AttributesImpl = {
+    val attrs = new AttributesImpl()
+    attrs.addAttribute(XMLUtils.XSI_NAMESPACE, "nil", "xsi:nil", "", "true")
+    attrs
+  }
+
+  /**
+   * Called by Daffodil internals to signify the beginning of an array of elements.
```

Comment:
Is this the same scaladoc as on a base trait/class? If so is there an idiom for avoiding repetition of the comment in the code while still getting the generated HTML documentation to have the same documentation generated?